syft/pkg/cataloger/golang/cataloger.go

```diff
@@ -5,28 +5,47 @@ package golang
 
 import (
 	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/event"
+	"github.com/anchore/syft/syft/pkg"
 	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/anchore/syft/syft/source"
 )
 
-type GoCatalogerOpts struct {
-	SearchLocalModCacheLicenses bool
-	LocalModCacheDir            string
-}
-
 // NewGoModFileCataloger returns a new Go module cataloger object.
-func NewGoModFileCataloger(opts GoCatalogerOpts) *generic.Cataloger {
+func NewGoModFileCataloger(opts GoCatalogerOpts) *ProgressingCataloger {
 	c := goModCataloger{
 		licenses: newGoLicenses(opts),
 	}
-	return generic.NewCataloger("go-mod-file-cataloger").
-		WithParserByGlobs(c.parseGoModFile, "**/go.mod")
+	return &ProgressingCataloger{
+		progress: c.licenses.progress,
+		cataloger: generic.NewCataloger("go-mod-file-cataloger").
+			WithParserByGlobs(c.parseGoModFile, "**/go.mod"),
+	}
 }
 
 // NewGoModuleBinaryCataloger returns a new Golang cataloger object.
-func NewGoModuleBinaryCataloger(opts GoCatalogerOpts) *generic.Cataloger {
+func NewGoModuleBinaryCataloger(opts GoCatalogerOpts) *ProgressingCataloger {
 	c := goBinaryCataloger{
 		licenses: newGoLicenses(opts),
 	}
-	return generic.NewCataloger("go-module-binary-cataloger").
-		WithParserByMimeTypes(c.parseGoBinary, internal.ExecutableMIMETypeSet.List()...)
+	return &ProgressingCataloger{
+		progress: c.licenses.progress,
+		cataloger: generic.NewCataloger("go-module-binary-cataloger").
+			WithParserByMimeTypes(c.parseGoBinary, internal.ExecutableMIMETypeSet.List()...),
+	}
+}
+
+type ProgressingCataloger struct {
```

Comment:
this looks like this is trying to be a common abstraction, maybe belongs in `generic`?